content/blog/2025-05-08-go-client-GA.md

```diff
@@ -0,0 +1,503 @@
+go mod tidy
+```
+
+Then, you can get started connecting to a Valkey standalone server, running locally on port 6379, with the following sample applications:
+
+```go
+package main
+
+import (
+    "fmt"
+    "github.com/valkey-io/valkey-glide/go/api"
+)
+
+func main() {
+    // Connect to a standalone Valkey server
+    config := api.NewGlideClientConfiguration().
+        WithAddress(&api.NodeAddress{Host: "localhost", Port: 6379})
+    
+    client, err := api.NewGlideClient(config)
+    if err != nil {
+        fmt.Println("Error:", err)
+        return
+    }
+    defer client.Close()
+    
+    // Test the connection
+    result, err := client.Ping()
+    if err != nil {
+        fmt.Println("Error:", err)
+        return
+    }
+    fmt.Println(result) // PONG
+    
+    // Store and retrieve a value
+    client.Set("hello", "valkey")
+    value, _ := client.Get("hello")
+    fmt.Println(value) // valkey
+}
+```
+
+### Cluster Mode Connection Setup
+
+Need to work with a Valkey cluster?
+
+Just as easy! The Go client automatically discovers your entire cluster topology from a single seed node. The following sample shows how to connect to a Valkey cluster through a node running locally on port 7001:
+
+```go
+package main
+
+import (
+    "fmt"
+    "github.com/valkey-io/valkey-glide/go/api"
+)
+
+func main() {
+    // Specify the address of any single node in your cluster
+    // This example connects to a local cluster node on port 7001
+    host := "localhost"
+    port := 7001
+    
+    // Connect to a Valkey cluster through any node
+    config := api.NewGlideClusterClientConfiguration().
+        WithAddress(&api.NodeAddress{Host: host, Port: port})
+    
+    client, err := api.NewGlideClusterClient(config)
+    if err != nil {
+        fmt.Println("There was an error: ", err)
+        return
+    }
+    
+    res, err := client.Ping()
+    if err != nil {
+        fmt.Println("There was an error: ", err)
+        return
+    }
+    fmt.Println(res) // PONG
+    client.Close()
+}
+```
+
+## Advanced Configuration Options
+
+### Read Strategies for Optimized Performance
+
+Balance consistency and throughput with flexible read strategies:
+
+```go
+// Configure to prefer replicas for read operations
+config := api.NewGlideClusterClientConfiguration().
+    WithAddress(&api.NodeAddress{Host: "cluster.example.com", Port: 6379}).
+    WithReadFrom(api.PreferReplica)
+
+client, err := api.NewGlideClusterClient(config)
+
+// Write to primary
+client.Set("key1", "value1")
+
+// Automatically reads from a replica (round-robin)
+result, err := client.Get("key1")
+```
+
+Available strategies:
+
+- **PRIMARY**: Always read from primary nodes for the freshest data
+- **PREFER_REPLICA**: Distribute reads across replicas in round-robin fashion, falling back to primary when needed
+
+Planned for future release:
+
+- **AZ_AFFINITY**: (Coming soon) Prefer replicas in the same availability zone as the client
```

Comment:
Is there an issue for this? I would want to back up somehow that this is planned beyond you saying it in this blog post. Also, provide a place where people can 👍 the issue or whatever.